packages/flutter/lib/src/widgets/container.dart

```diff
@@ -374,6 +374,9 @@ class Container extends StatelessWidget {
     }
     final EdgeInsetsGeometry? decorationPadding = decoration!.padding;
     if (padding == null) {
+      if (decorationPadding == EdgeInsets.zero) {
```

Comment:
Is this a separate change from deprecating BoxDecoration.shape?

Reply:
It can be separate. I was trying to get the tests working and found ShapeDecoration adds padding ALWAYS, while BoxDecoration doesn't. This makes ShapeDecoration make one less widget and be on par with BoxDecoration.

There is zero difference, just one less widget on the tree.

Reply:
This broke a really random test :( reverted.